From 4fb2810add2ad1eb6199ae1dd80fc70adf620d8d Mon Sep 17 00:00:00 2001
From: Lukas Reschke <lukas@statuscode.ch>
Date: Fri, 19 Aug 2016 14:46:20 +0200
Subject: [PATCH] Remove reading PATH_INFO from server variable

Having two code paths for this is unreliable and can lead to bugs. Also, in some cases Apache isn't setting the PATH_INFO variable when mod_rewrite is used.

Fixes https://github.com/nextcloud/server/issues/983
---
 lib/private/AppFramework/Http/Request.php   |  4 ----
 tests/lib/AppFramework/Http/RequestTest.php | 16 ----------------
 2 files changed, 20 deletions(-)

--- a/lib/private/AppFramework/Http/Request.php
+++ b/lib/private/AppFramework/Http/Request.php
@@ -701,10 +701,6 @@ class Request implements \ArrayAccess, \
 	 * @return string|false Path info or false when not found
 	 */
 	public function getPathInfo() {
-		if(isset($this->server['PATH_INFO'])) {
-			return $this->server['PATH_INFO'];
-		}
-
 		$pathInfo = $this->getRawPathInfo();
 		// following is taken from \Sabre\HTTP\URLUtil::decodePathSegment
 		$pathInfo = rawurldecode($pathInfo);
